### PR TITLE
feat(cli): warn when recursive type depth not set in prism schema

### PR DESCRIPTION
**Pull request by leejayhsu**
## Change Summary

Warn users when they don't set recursive type depth in the prism schema. The warning should tell users that they can set it to -1 for use with Pyright.

Closes https://github.com/RobertCraigie/prisma-client-py/issues/252

## Checklist

- [x] Unit tests for the changes exist
- [x] Tests pass without significant drop in coverage
- [x] Documentation reflects changes where applicable
- [x] Test snapshots have been [updated](https://prisma-client-py.readthedocs.io/en/latest/contributing/contributing/#snapshot-tests) if applicable

## Agreement

By submitting this pull request, I confirm that you can use, modify, copy and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/prisma/generator/models.py [hunk 1]
```diff
@@ -462,7 +481,15 @@ def partial_type_generator_converter(
...
     @classmethod
-    def recursive_type_depth_validator(cls, value: int) -> int:
+    def recursive_type_depth_validator(cls, value: Optional[int]) -> int:
+        if value is None:
+            click.echo(
+                click.style(
+                    RECURSIVE_TYPE_DEPTH_WARNING,
+                    fg='yellow',
+                )
+            )
```
Comment (RobertCraigie) [inline src/prisma/generator/models.py:491]: It looks nicer to me if just the first line of the warning is highlighted yellow and the rest of the warning is left to the default. Although that could make it look like two different messages if we ever add another warning.

What do you think?

**Review by RobertCraigie — CHANGES_REQUESTED** [review]
Thank you! 💜 

Just some minor changes and nitpicks.

File: src/prisma/generator/models.py [hunk 2]
```diff
@@ -372,7 +391,7 @@ class Config(BaseSettings):
 
     interface: InterfaceChoices = InterfaceChoices.asyncio
     partial_type_generator: Optional[Module] = None
-    recursive_type_depth: int = FieldInfo(default=5)
+    recursive_type_depth: Optional[int] = None
```
[the remark this answers is not in the document]
Comment (leejayhsu) [reply src/prisma/generator/models.py]: totally agree! done in https://github.com/RobertCraigie/prisma-client-py/pull/386/commits/1fd2ffcb0fecea34bb194c146382a4899841cd41

File: tests/test_generation/test_models.py [hunk 3]
```diff
@@ -32,3 +32,21 @@ def test_recursive_type_depth() -> None:
...
+
+    Ensure that we provide advice on what value to use.
+    Also ensure that it defaults to 5.
+    """
+    c = Config()
+    captured = capsys.readouterr()
+    assert 'it is highly recommended to use Pyright' in captured.out.replace(
+        '\n', ' '
+    )
+    assert c.recursive_type_depth == 5
```
[the remark this answers is not in the document]
Comment (leejayhsu) [reply tests/test_generation/test_models.py:53]: done in https://github.com/RobertCraigie/prisma-client-py/pull/386/commits/684c870a4fa0007cd16f61556cf0b33e8e78a67d

File: src/prisma/generator/models.py (hunk 1 above)
Comment (leejayhsu) [reply src/prisma/generator/models.py]: yeah all the yellow text can be quite jarring. After playing around with colors, I agree that highlighting the first part of the message is probably the best compromise.

I guess we can worry about how it looks with other warnings when it becomes a problem?

**Review by RobertCraigie — APPROVED** [review]
Looks great, thank you! :purple_heart:

File: src/prisma/generator/models.py (hunk 1 above)
Comment (RobertCraigie) [reply src/prisma/generator/models.py]: Yeah that's a good point, we can worry about that later.

**Pull request merged**